internal/template/template.go

```diff
@@ -256,6 +260,40 @@ func (t *Templator) execTemplate(name string, tmplContent string, data any, repl
 	return buf.String(), nil
 }
 
+func recurseCanary() []string {
+	return []string{
```

Comment:
does this mean we only have a maximum of 5 recursions? Prob not an issue but maybe another way of doing this that is not restricted?